src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVKmerizerWithLowComplexityFilter.java

```diff
@@ -0,0 +1,90 @@
+    private final double minEntropy;
+    private final int[] baseCounts;
+    private static double[] entropies;
+
+    public SVKmerizerWithLowComplexityFilter( final byte[] seq, final int kSize, final double minEntropy ) {
+        this(new ASCIICharSequence(seq), kSize, minEntropy);
+    }
+
+    public SVKmerizerWithLowComplexityFilter( final CharSequence seq, final int kSize, final double minEntropy ) {
+        super(kSize, seq);
+        this.minEntropy = minEntropy;
+        baseCounts = new int[Base.values().length];
+        baseCounts[(int)Base.A.value] = kSize;
+        this.nextKmer = nextKmer(new SVKmer(kSize), 0);
+    }
+
+    public static Stream<SVKmer> stream( final CharSequence seq, final int kSize, final double minEntropy ) {
+        return StreamSupport.stream(((Iterable<SVKmer>)() ->
+                new SVKmerizerWithLowComplexityFilter(seq, kSize, minEntropy)).spliterator(), false);
+    }
+
+    public static Stream<SVKmer> stream( final byte[] seq, final int kSize, final double minEntropy ) {
+        return StreamSupport.stream(((Iterable<SVKmer>)() ->
+                new SVKmerizerWithLowComplexityFilter(seq, kSize, minEntropy)).spliterator(), false);
+    }
+
+    protected SVKmer nextKmer( SVKmer tmpKmer, int validBaseCount ) {
+        final double[] entropies = getEntropies(kSize);
+        final int len = seq.length();
+        while ( idx < len ) {
+            final int firstIdx = (int)tmpKmer.firstBase(kSize).value;
+            switch ( seq.charAt(idx) ) {
+                case 'a': case 'A':
+                    tmpKmer = tmpKmer.successor(Base.A, kSize);
+                    baseCounts[firstIdx] -= 1;
+                    baseCounts[(int)Base.A.value] += 1;
+                    break;
+                case 'c': case 'C':
+                    tmpKmer = tmpKmer.successor(Base.C, kSize);
+                    baseCounts[firstIdx] -= 1;
+                    baseCounts[(int)Base.C.value] += 1;
+                    break;
+                case 'g': case 'G':
+                    tmpKmer = tmpKmer.successor(Base.G, kSize);
+                    baseCounts[firstIdx] -= 1;
+                    baseCounts[(int)Base.G.value] += 1;
+                    break;
+                case 't': case 'T':
+                    tmpKmer = tmpKmer.successor(Base.T, kSize);
+                    baseCounts[firstIdx] -= 1;
+                    baseCounts[(int)Base.T.value] += 1;
+                    break;
+                default:
+                    validBaseCount = -1;
+                    break;
+            }
+            idx += 1;
+
+            if ( ++validBaseCount >= kSize ) {
+                final double entropy = Arrays.stream(baseCounts).mapToDouble(count -> entropies[count]).sum();
+                if ( entropy >= minEntropy ) return tmpKmer;
+            }
+        }
+        return null;
+    }
+
+    private static double[] getEntropies( final int kSize ) {
```

Comment:
Can you add a comment explaining this? Ie something like "This returns an array where the i'th entry is the contribution to the total entropy of a kmer of having i occurrences of a single base a kmer of the given size"..